sbt-metals/src/main/scala/scala/meta/metals/MetalsPlugin.scala

```diff
@@ -48,6 +48,20 @@ object MetalsPlugin extends AutoPlugin {
         )
     },
     javaSemanticdbEnabled := bspEnabled.value,
+    javaHome := {
+      javaHome.value match {
+        case None =>
+          // In case if jdk is >= 17 we need to set `javaHome` explicitly
+          // to force sbt creating ForkJavaCompiler.
+          // Otherwise it creates LocalJavaCompiler that ignores `-J` flags
+          // and semanticdb plugin fails compilation
```

Comment:
Can you leave a link to

- https://github.com/sbt/zinc/blob/dd1f3596494b4c4d7a19256bc921e765d3dc12c8/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/JavaCompiler.scala#L41-L67
- and https://github.com/sbt/zinc/blob/dd1f3596494b4c4d7a19256bc921e765d3dc12c8/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/javac/LocalJava.scala#L299-L304 ?

At first, I couldn't understand who is `ForkJavaCompiler` and `LocalJavaCompiler`.

Reply:
added a link